### DISCUSSION TITLE
How to get the Markdown pages with their frontmatter efficiently?

### DISCUSSION THREAD

**Discussion by shamscorner** [antfu/vitesse#186] [Q&A]
I can see how this repo reads all the locales from the `locales/` directory in the `modules/i18n.ts`, 
```js
const messages = Object.fromEntries(
  Object.entries(
    import.meta.globEager('../../../locales/*.y(a)?ml'))
    .map(([key, value]) => {
      const yaml = key.endsWith('.yaml')
      return [key.slice(17, yaml ? -5 : -4), value.default]
    }),
)
````
However, is there any way to get all the markdown files and their frontmatter effectively? I want to create a blog homepage. And for that, I have to get all the contents like *Nuxt* so that I can show the title, description, image, date, etc.

Thanks for your time.

**Comment by antfu** [comment]
https://github.com/antfu/vite-plugin-md#frontmatter

**Reply by shamscorner** [reply]
> https://github.com/antfu/vite-plugin-md#frontmatter

@antfu Yes, I already checked that out. But the thing is, I want all the markdowns and their frontmatters like a JSON form so that I can make a homepage for all the blog posts. Can I do that?
